src/modules/Invoice/Service.php

```diff
@@ -511,7 +511,7 @@ public function setInvoiceDefaults(\Model_Invoice $model)
 
         $model->serie = $systemService->getParamValue('invoice_series');
         $model->nr = $model->id;
-        $model->hash = hash('sha256', random_bytes(13));
+        $model->hash = bin2hex(random_bytes(random_int(100, 127)));;
```

Comment:
Still limited to 0-9a-e would prefer to see something like a-zA-Z0-9

Reply:
That does not at all matter here. This is already overkill